### PR TITLE
Material: Material appearance

### PR DESCRIPTION
**Pull request by chennes**
The original PR, #13024, is beyond my ability to fix -- the integrated formatting changes result in too many conflicts. This is a clean rebuild of that PR's contents, without the formatting changes.

----

Uses new material system for appearance

Each feature object now has a property called ShapeMaterial that describes its physical properties. If it has a shape, it has a material.

The ShapeColor attribute is replaced by a ShapeAppearance attribute. This is a material list that describes all appearance properties, not just diffuse color. As a list in can be used for all elements of a shape, such as edges and faces.

A new widget is provided to allow the user to select materials in a consistent fashion. It can also launch the material editor with its more advanced capabilities.

### CODE REVIEW

**Comment by andir** [comment]
This seems to have broken the CAM simluator. I'm now getting exceptions such as below with revision 
ee39814f0c1ae96de446ac96070c9d0eb277128f
```
11:32:11  Running the Python command 'CAM_Simulator' failed:
Traceback (most recent call last):
  File "[...]/Mod/CAM/Path/Main/Gui/Simulator.py", line 647, in Activated
    pathSimulation.Activate()
  File "[...]/Mod/CAM/Path/Main/Gui/Simulator.py", line 111, in Activate
    self.SimulateMill()
  File "[...]/Mod/CAM/Path/Main/Gui/Simulator.py", line 231, in SimulateMill
    self.cutMaterialIn.ViewObject.ShapeAppearance.DiffuseColor = (1.0, 0.85, 0.45, 0.0)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

'tuple' object has no attribute 'DiffuseColor'
```

File: src/Mod/Material/Gui/DlgMaterialImp.h [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+namespace MatGui
+{
+
+class ViewProvider;
+class Command;
+
+/**
+ * The DlgMaterialImp class implements a dialog containing all available document
+ * templates to create a new document.
+ * \author Jürgen Riegel
```
Comment (0penBrain) [inline src/Mod/Material/Gui/DlgMaterialImp.h:52]: What does this mean ?

File: src/Mod/Material/Gui/DlgDisplayPropertiesImp.h [hunk 2]
```diff
@@ -1,131 +1,136 @@
...
+namespace MatGui
+{
+
+class ViewProvider;
+class Command;
+
+/**
+ * The DlgDisplayPropertiesImp class implements a dialog containing all available document
+ * templates to create a new document.
+ * \author Jürgen Riegel
```
Comment (0penBrain) [inline src/Mod/Material/Gui/DlgDisplayPropertiesImp.h:52]: What does this mean ?

File: src/Mod/Material/Gui/DlgDisplayProperties.ui [hunk 3]
```diff
@@ -434,6 +332,101 @@
...
+         <widget class="QLabel" name="label_4">
+          <property name="text">
+           <string>Color plot:</string>
+          </property>
+         </widget>
+        </item>
+        <item row="0" column="0">
+         <widget class="QLabel" name="label_6">
+          <property name="text">
+           <string>Shape Appearance:</string>
```
Comment (0penBrain) [inline src/Mod/Material/Gui/DlgDisplayProperties.ui:372]: Is it known and being fixed that "Shape Appearance" dialog is completely broken and do strictly nothing ?

**Pull request merged**